uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/WorkPatternCalendarTest.php

```diff
@@ -456,4 +396,170 @@ public function testItGeneratesAccurateNumberOfCalendarDatesUsingOnlyTheContract
     //Year 2016 is a leap year, so there are 366 days.
     $this->assertCount(366, $calendarDates);
   }
+
+  public function testGetReturnsResultsForContactWithoutAContractAndFillsTheContactWorkPatternPeriodLapsesWithTheDefaultWorkPattern() {
+    $absencePeriod = AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('2016-03-03'),
+      'end_date' => CRM_Utils_Date::processDate('2016-03-09'),
+    ]);
+
+    WorkPatternFabricator::fabricateWithA40HourWorkWeek(['is_default' => true]);
+    $workPattern1 = WorkPatternFabricator::fabricateWithTwoWeeksAnd31AndHalfHours();
+
+    ContactWorkPatternFabricator::fabricate([
+      'contact_id' => $this->contact['id'],
+      'pattern_id' => $workPattern1->id,
+      'effective_date' => CRM_Utils_Date::processDate('2016-03-07'),
+    ]);
+
+    //A contract is fabricated for the contract with start date as 2016-03-03 and end date as 2016-03-09
```

Comment:
Saying "A contract is fabricated for the contract" doesn't make much sense to me. Maybe "A contract is fabricated for the period betwen dateX and dateY" would be better?